src/components/project/project-member/project-member.service.ts

```diff
@@ -279,64 +287,62 @@ export class ProjectMemberService {
   }
 
   async readOne(id: string, session: ISession): Promise<ProjectMember> {
-    if (!session.userId) {
-      session.userId = this.config.anonUser.id;
+    this.logger.info(`Query readOne ProjectMember`, {
+      id,
+      userId: session.userId,
+    });
+    if (!id) {
+      throw new BadRequestException('No project memeber id to search for');
     }
-
-    const props = ['roles', 'modifiedAt'];
-
-    const baseNodeMetaProps = ['id', 'createdAt'];
-
-    const childBaseNodeMetaProps: ChildBaseNodeMetaProperty[] = [
-      {
-        parentBaseNodePropertyKey: 'user',
-        parentRelationDirection: 'out',
-        childBaseNodeLabel: 'User',
-        childBaseNodeMetaPropertyKey: 'id',
-        returnIdentifier: 'userId',
-      },
-    ];
-
     const query = this.db
       .query()
       .call(matchRequestingUser, session)
-      .call(matchUserPermissions, 'ProjectMember', id)
-      .call(addAllSecureProperties, ...props)
-      .call(addAllMetaPropertiesOfChildBaseNodes, ...childBaseNodeMetaProps)
-      .with([
-        ...props.map(addPropertyCoalesceWithClause),
-        ...childBaseNodeMetaProps.map(addShapeForChildBaseNodeMetaProperty),
-        ...baseNodeMetaProps.map(addShapeForBaseNodeMetaProperty),
-        'node',
+      .match([node('node', 'ProjectMember', { active: true, id })])
+      .call(getPermList, 'requestingUser')
+      .call(getPropList, 'permList')
+      .match([
+        node('node'),
+        relation('out', '', 'user'),
+        node('user', 'User', { active: true }),
       ])
-      .returnDistinct([
-        ...props,
-        ...baseNodeMetaProps,
-        ...childBaseNodeMetaProps.map((x) => x.returnIdentifier),
-        'labels(node) as labels',
-      ]);
+      .return('node, permList, propList, user.id as userId')
+      .asResult<
+        StandardReadResult<DbPropsOfDto<ProjectMember>> & {
+          userId: string;
+        }
+      >();
 
     const result = await query.first();
+
     if (!result) {
       throw new NotFoundException('Could not find project memeber');
     }
 
-    const response: any = {
-      ...result,
-      roles: {
-        value: result.roles.value || [],
-        canRead: result.roles.canRead,
-        canEdit: result.roles.canEdit,
-      },
-      modifiedAt: result.modifiedAt.value,
+    const securedProps = parseSecuredProperties(
+      result.propList,
+      result.permList,
+      {
+        roles: true,
+        modifiedAt: true,
+      }
+    );
+
+    const perms = parsePermissions(result.permList);
+
+    return {
+      ...parseBaseNodeProperties(result.node),
+      ...securedProps,
       user: {
         value: await this.userService.readOne(result.userId, session),
-        canRead: !!result.canReadUser,
-        canEdit: !!result.canEditUser,
+        canRead: !!(perms as any).user?.canRead,
+        canEdit: !!(perms as any).user?.canEdit,
```

Comment:
We should be able to handle these types for this case without using `as any`. Have you tried using `user: true` on `parseSecureProperties`? If so, I'll take a look tomorrow.

Reply:
HI @CarsonF Please take a look at and let me know